src/internal/visual-mode/index.ts

```diff
@@ -12,7 +12,7 @@ import { awsuiVisualRefreshFlag, getGlobal } from '../global-flags';
 export function isMotionDisabled(element: HTMLElement): boolean {
   return (
     !!findUpUntil(element, node => node.classList.contains('awsui-motion-disabled')) ||
-    (window.matchMedia?.('(prefers-reduced-motion: reduce)').matches ?? false)
+    (element?.ownerDocument?.defaultView?.matchMedia?.('(prefers-reduced-motion: reduce)')?.matches ?? false)
```

Comment:
not being able to pinpoint the exact cause of the issue, I think being ultra safe wouldn't hurt. 

do you see any concerns with that?

Reply:
I would also add a comment or maybe replace optional chaining with a try catch as there is no test to secure the change and we can remove that in some future refactoring as potentially unnecessary.

The ownerDocument and defaultView are not really expected to be null and when it is so it might indicate a problem. To the very least, the current util might produce wrong result. I think issuing a console warning might be helpful for debugging.

Reply:
@abdhalees to the very least I would remove the `?` after element. We only use that function in our own components and if we pass a null or undefined despite the type suggests otherwise - that is an error that needs to be thrown.